lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -125,6 +125,20 @@ using Scheduleable =
 
 class IfLoweringStateInterface {
 public:
+  void setCondReg(scf::IfOp op, calyx::RegisterOp regOp) {
+    Operation *operation = op.getOperation();
+    assert(condReg.count(operation) == 0 &&
+           "A condition register was already set for this scf::IfOp!\n");
+    condReg[operation] = regOp;
```

Comment:
To avoid the double lookup here (once for `count` and once for the `[]` operator, we can do:

```cpp
auto [it, succeeded] = condReg.insert(std::make_pair(operation, regOp));
assert(succeeded && "...");
```